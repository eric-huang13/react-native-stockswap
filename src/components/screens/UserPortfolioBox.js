import React, { Component } from 'react'
import { Text, View, StyleSheet } from 'react-native'
import PortfolioGraph from './PortfolioGraph'

export default class UserPortfolioBox extends Component {
    constructor(props) {
        super(props);
        this.state = {
          graphData: [
            {x: 2, y: 10},
            {x: 3, y: 11},
            {x: 4, y: 12},
            {x: 5, y: 14},
            {x: 6, y: 14},
            {x: 7, y: 15},
          ],
          percent: '1.22',
          range: [10, 15],
         
        };
      }
    
    render() {

        const {item} = this.props
        const {graphData, percent, range} = this.state;
        console.log(item,'item in portfolioBOX')

        return (
            <View style={style.container}>
                <View style={style.symbolContainer}>
                    <Text style={style.symbol}>{item.symbol}</Text>
                    <Text style={style.title}>{item.title}</Text>
                    <Text style={style.price}>Price: ${item.price}</Text>
                </View>
                <View style={style.graphContainer}>
                <PortfolioGraph graphData={graphData} range={range} />
                {/* <Text>graph</Text> */}

                </View>
                <View style={style.percentContainer}>
                <Text style={style.percent}>{percent}</Text>
                </View>
            </View>
        )
    }
}

const style = StyleSheet.create({
container:{
        backgroundColor: '#2a334a',
        flexDirection:'row',
        // borderBottomWidth:1,
        justifyContent:'space-between',
        // paddingHorizontal:2,
        // paddingVertical:20,
        flex:1,
        // borderWidth:2,
        // borderColor:'lightgrey',
        // margin:4,
        width:99,
        // backgroundColor: "#eee",
    // marginHorizontal: 20,
    // paddingHorizontal: 2,
    // paddingBottom: 80,
    // paddingTop: 10,
    // borderRadius: 10,
    // height: "88%",
        
    },
symbolContainer:{
        flexDirection:'column',
        justifyContent:'space-evenly',
    // borderBottomWidth:.5,
    // borderColor:'lightgrey',
    paddingVertical:4,
    paddingLeft:8,

    width:"100%",
    // height:'100%',
    // alignItems:'le',
    // alignContent:'left',
    textAlign:'left'
    },
symbol:{
        color:'#91f2b1',
        fontWeight:'bold',
        fontSize:18,
        textAlign:'left',               

    },
title:{
        color:'lightgrey',
        fontSize:12,
        textAlign:'left',      

    },
price:{
        color:'white',
        textAlign:'left',  
        fontSize:13,
      

    },
    graphContainer:{
        // borderWidth:.5,
        // borderColor:'lightgrey',
        // paddingBottom:1
    },
    percentContainer:{       
        justifyContent:'center'

    },
    percent:{
        color:'#91f2b1',
        fontWeight:'bold',
        fontSize:17,
        textAlign:'left',
    },


})
