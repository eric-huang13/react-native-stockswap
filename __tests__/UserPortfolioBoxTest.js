import React from 'react';
import UserPortfolioBox from '../src/components/HomeTabComponents/UserPortfolioBox';
import ShallowRenderer from 'react-test-renderer/shallow'; // ES6


jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));
  

test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render   (<UserPortfolioBox item={[]} />);
    const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});