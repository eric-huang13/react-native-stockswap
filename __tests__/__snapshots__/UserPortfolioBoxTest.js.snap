// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#2a334a",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "width": 139.8284671532847,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "column",
        "justifyContent": "space-evenly",
        "paddingLeft": 11.299270072992702,
        "paddingVertical": 5.649635036496351,
        "textAlign": "left",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#71F59C",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 22.598540145985403,
          "textAlign": "left",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "lightgrey",
          "fontFamily": "Montserrat-Regular",
          "fontSize": 14.124087591240876,
          "textAlign": "left",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Montserrat-Regular",
          "fontSize": 16.94890510948905,
          "textAlign": "left",
        }
      }
    >
      Price: $
    </Text>
  </View>
  <View>
    <PortfolioGraph
      graphData={
        Array [
          Object {
            "x": 2,
            "y": 10,
          },
          Object {
            "x": 3,
            "y": 21,
          },
          Object {
            "x": 4,
            "y": 32,
          },
          Object {
            "x": 5,
            "y": 14,
          },
          Object {
            "x": 6,
            "y": 4,
          },
          Object {
            "x": 7,
            "y": 25,
          },
        ]
      }
      percent="1.22"
      range={
        Array [
          5,
          30,
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flexDirection": "column",
        "justifyContent": "space-evenly",
        "paddingRight": 42.372262773722625,
        "paddingVertical": 5.649635036496351,
      }
    }
  >
    <BullIcon
      style={
        Object {
          "alignSelf": "flex-end",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#71F59C",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 22.598540145985403,
          "textAlign": "left",
        }
      }
    >
      1.22
      %
    </Text>
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Montserrat-Regular",
          "fontSize": 16.94890510948905,
          "textAlign": "left",
        }
      }
    >
      Portfolio:
    </Text>
  </View>
</ForwardRef(SafeAreaView)>
`;
